docs/azure/sdk/dependency-injection.md

```diff
@@ -25,34 +35,32 @@ dotnet add package Azure.Security.KeyVault.Secrets
 dotnet add package Azure.Storage.Blobs
 ```
 
-## Register client
+## Register clients
 
-In the _Program.cs_ file, register a client for each service:
+In the *Program.cs* file, invoke the <xref:Microsoft.Extensions.Azure.AzureClientServiceCollectionExtensions.AddAzureClients%2A> extension method to register a client for each service. The following code samples provide guidance on application builders from the `Microsoft.AspNetCore.Builder` and `Microsoft.Extensions.Hosting` namespaces.
 
-```csharp
-builder.Services.AddAzureClients(clientBuilder =>
-{
-    // Add a KeyVault client
-    clientBuilder.AddSecretClient(keyVaultUrl);
+### [WebApplicationBuilder](#tab/web-app-builder)
 
-    // Add a Storage account client
-    clientBuilder.AddBlobServiceClient(storageUrl);
+:::code language="csharp" source="snippets/dependency-injection/WebApplicationBuilder/Program.cs" id="snippet_WebApplicationBuilder" highlight="6-11":::
 
-    // Use DefaultAzureCredential by default
-    clientBuilder.UseCredential(new DefaultAzureCredential());
-});
+### [HostApplicationBuilder](#tab/host-app-builder)
 
-builder.Services.AddControllers();
-```
+:::code language="csharp" source="snippets/dependency-injection/HostApplicationBuilder/Program.cs" highlight="7-12":::
+
+### [HostBuilder](#tab/host-builder)
+
+:::code language="csharp" source="snippets/dependency-injection/HostBuilder/Program.cs" id="snippet_HostBuilder" highlight="8-13":::
+
+---
```

Comment:
Do we have a doc somewhere that summarizes the differences between these builders or the different builder versions? I tried to find one but couldn't, just thought it might be useful.

Reply:
The `Host.CreateDefaultBuilder` is the older version, the new (minimal) approach is to use `Host.CreateApplicationBuilder`. That's where our docs leaning.

Reply:
Right, I just meant that users looking at these tabs might not know that or might not know which version they should be using unless we give them a link to more information. I guess we could assume that if they're reading this they already have an app setup and will pick the one that matches their program.cs.